src/PerformanceTests/NServiceBus6/Session.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Threading.Tasks;
+using NServiceBus;
+
+class Session : ISession
+{
+    readonly IEndpointInstance instance;
+
+    public Session(IEndpointInstance instance)
+    {
+        this.instance = instance;
+    }
+
+    public Task Send(object message)
+    {
+        return instance.Send(message);
```

Comment:
I suggest you watch my webinar series ;)

In this case it is better to return. No statemachine creation = less memory
and no context capturing

On Monday, April 18, 2016, Ramon Smits <notifications@github.com> wrote:

> In src/PerformanceTests/NServiceBus6/Session.cs
> <https://github.com/Particular/EndToEnd/pull/61#discussion_r60098544>:
>
> > @@ -0,0 +1,27 @@
> > +﻿using System.Threading.Tasks;
> > +using NServiceBus;
> > +
> > +class Session : ISession
> > +{
> > +    readonly IEndpointInstance instance;
> > +
> > +    public Session(IEndpointInstance instance)
> > +    {
> > +        this.instance = instance;
> > +    }
> > +
> > +    public Task Send(object message)
> > +    {
> > +        return instance.Send(message);
>
> @danielmarbach <https://github.com/danielmarbach> This is allowed
> correct? It is not required to do:
>
> public async Task Send(object message)
> {
>     await instance.Send(message);
> }
>
> It feels strange to await when you can just return but I don't know if
> this is bad practice and if it is, why its so.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/EndToEnd/pull/61/files/abd2cd5b5e4122dd62555d09603fd491f3d16e91#r60098544>
>